### PR TITLE
Fix issue in ADURLProtocol when URLs get canonicalized

### PR DESCRIPTION
**Pull request by RPangrle**
Remove  [self.client URLProtocol:self wasRedirectedToRequest:mutableRequest redirectResponse:response] call. This would cause multiple requests to the same endpoint that could break some configurations.

Replace the @"ADURLProtocol" string with a constant.

### CODE REVIEW

File: ADAL/src/ui/ADWebAuthController.m [hunk 1]
```diff
@@ -417,6 +417,19 @@ - (void)webAuthDidFailWithError:(NSError *)error
     {
         return;
     }
+    
+    // If we failed on an invalid URL check to see if it matches our end URL
+    if ([error.domain isEqualToString:@"NSURLErrorDomain"] && error.code == -1002)
+    {
+        NSURL* url = [error.userInfo objectForKey:NSURLErrorFailingURLErrorKey];
+        if ([[[url absoluteString] lowercaseString] hasPrefix:_endURL])
```
Comment (jasoncoolmax) [inline ADAL/src/ui/ADWebAuthController.m]: should be [_endURL lowercaseString]? :)

File: ADAL/src/urlprotocol/ADURLProtocol.m [hunk 2]
```diff
@@ -202,36 +213,38 @@ - (void)connection:(NSURLConnection *)connection
...
     (void)connection;
     
-    AD_LOG_VERBOSE_F(@"-[ADURLProtocol connection:willSendRequest:]", _correlationId, @"Redirect response: %@. New request:%@", response.URL, request.URL);
+    AD_LOG_VERBOSE_F(@"-[ADURLProtocol connection:willSendRequest:]", _correlationId, @"Redirect response: %@. New request:%@", redirectResponse.URL, request.URL);
     
     // Disallow HTTP for ADURLProtocol
     if ([request.URL.scheme isEqualToString:@"http"])
     {
-        if (!s_endURL)
+        if ([request.URL.absoluteString.lowercaseString hasPrefix:s_endURL])
```
Comment (jasoncoolmax) [inline ADAL/src/urlprotocol/ADURLProtocol.m]: Seems here it is only targetting at end URL in the form of "http://****", 
are we supposed to return error for all kinds of end url? :P

Comment (RPangrle) [inline ADAL/src/urlprotocol/ADURLProtocol.m]: non http/https can't be loaded in Safari anyways, and https URLs don't have the same security implications http ones do.

**Comment by jasoncoolmax** [comment]
:shipit:

**Pull request merged**